src/clj/schema/macros.clj

```diff
@@ -362,6 +362,155 @@
              (new ~(symbol (str name))
                   ~@(map (fn [s] `(safe-get ~map-sym ~(keyword s))) field-schema)))))))
 
+(defn -instrument-protocol-method
+  "Given a protocol Var pvar, its method method-var and instrument-method,
+  instrument the protocol method."
+  [pvar ;:- Var
+   method-var ;:- (Var InnerMth)
+   instrument-method #_:- #_(s/=>* OuterMth
+                                   [InnerMth
+                                    (named (=> Any OuterMth InnerMth)
+                                           'sync!)])]
+  (let [;; propagate method cache to inner method
+        sync! (fn [^clojure.lang.AFunction outer-mth
+                   ^clojure.lang.AFunction inner-mth]
+                (when-not (identical? (.__methodImplCache outer-mth)
```

Comment:
I don't know anything about this, is it maybe worth a bit more explanation of what's going on here (and/or a link to somewhere that explains the underlying semantics)?

Reply:
Added a comment block just above `sync!`.